modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -723,10 +731,10 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
         Halide::Func top = (name.empty() ? Halide::Func() : Halide::Func(name));
         Halide::RDom r(0, kernel.width, 0, kernel.height);
         Halide::Expr kx, ky;
-        if (pad.width || pad.height)
+        if(pad_l || pad_t || pad_r || pad_b)
         {
-            kx = clamp(x * stride.width + r.x - pad.width, 0, inWidth - 1);
-            ky = clamp(y * stride.height + r.y - pad.height, 0, inHeight - 1);
+            kx = clamp(x * stride.width + r.x - pad_l, 0, inWidth - 1);
+            ky = clamp(y * stride.height + r.y - pad_t, 0, inHeight - 1);
```

Comment:
Should be
```cpp
if(pad_l || pad_t)
{
```
because it checks for negative values in `x * stride.width + r.x - pad_l` or `y * stride.height + r.y - pad_t`.